enterprise/frontend/src/metabase-enterprise/google_drive/GdriveAddDataPanel.tsx

```diff
@@ -173,14 +221,19 @@ export const GdriveAddDataPanel = () => {
   return (
     <PanelWrapper
       isModalOpen={isConnectionModalOpen}
-      onModalClose={closeConnectionModal}
+      // We are closing the parent (Add data) modal so we don't have to explicitly
+      // close the connection modal as well. It will be removed from DOM with its parent gone.
+      onModalClose={onAddDataModalClose}
     >
       <Button
         variant="filled"
         w={INNER_WIDTH}
         onClick={() => {
           trackSheetConnectionClick({ from: "add-data-modal" });
-          openConnectionModal();
+
+          status === "not-connected"
+            ? openConnectionModal()
+            : showConnectionDetails();
```

Comment:
This is probably the key change in the PR.
We either show the (standalone) connection modal in order to create a new connection OR we show the connection details (inline, incorporated into the current panel)